test/libponyc/badpony.cc

```diff
@@ -1287,3 +1287,21 @@ TEST_F(BadPonyTest, CantAssignErrorExpression)
   TEST_ERRORS_1(src,
     "right side must be something that can be assigned")
 }
+
+TEST_F(BadPonyTest, NotSafeToWrite)
+{
+  // From issue #4290
+  const char* src =
+    "class Foo\n"
+    "class X\n"
+      "var foo: Foo ref = Foo\n"
+
+    "actor Main\n"
+      "new create(env: Env) =>\n"
+        "let x = X\n"
+        "let foo: Foo ref = Foo\n"
+        "x.foo = foo";
+
+  TEST_ERRORS_1(src,
+    "not safe to write right side to left side")
+}
```

Comment:
The test should really be covering all the errors. As it stands, this is the current message, so if someone was to make a change back to the "vague message", this test wouldn't catch it.

Reply:
You want to use: 

`test_expected_error_frames`

you can see its usage in `PartialFunCallInTryBlock` in test/libponyc/verify.cc at line 672.

Reply:
Done